torch/_functorch/eager_transforms.py

```diff
@@ -1660,15 +1668,15 @@ def forward_ad_checks(flat_tangents):
     #   calling the folded fx graph and unflattening fx graph output
     def jvp_fn(*tangents):
         flat_tangents, tangent_argspec = tree_flatten(tangents)
-        if tangent_argspec != primals_argspec:
-            raise RuntimeError(f"Expected the tangents {tangent_argspec} to have "
-                               f"the same argspec as the primals {primals_argspec}")
+        _linearize_treespec_compare(primals, tangents)
+        # if tangent_argspec != primals_argspec:
+        #     raise RuntimeError(f"Expected the tangents {tangent_argspec} to have "
+        #                        f"the same argspec as the primals {primals_argspec}")
```

Comment:
delete?

Reply:
mpile_check(wrapper_fn, (x,), fullgraph=False, graph_idx=1)
+        actual = normalize_gm(wrapped_gm.print_readable(print_output=False))
+        self.assertExpectedInline(
+            actual,
+            """\
+class GraphModule(torch.nn.Module):
+    def forward(self, getattr_L_self_FX_CONST_FOLDED_ATTRS_0_ : torch.nn.parameter.Parameter, getattr_L_self_FX_CONST_FOLDED_ATTRS_1_ : torch.nn.parameter.Parameter, L_flat_tangents_1_ : torch.Tensor):
+        getattr_l_self_fx_const_folded_attrs_0_ = getattr_L_self_FX_CONST_FOLDED_ATTRS_0_
+        getattr_l_self_fx_const_folded_attrs_1_ = getattr_L_self_FX_CONST_FOLDED_ATTRS_1_
+        l_flat_tangents_1_ = L_flat_tangents_1_
+
+        _new_zeros_with_same_feature_meta_default = torch.ops.aten._new_zeros_with_same_feature_meta.default(l_flat_tangents_1_, getattr_l_self_fx_const_folded_attrs_0_);  getattr_l_self_fx_const_folded_attrs_0_ = None
+
+        copy__default = torch.ops.aten.copy_.default(_new_zeros_with_same_feature_meta_default, l_flat_tangents_1_);  _new_zeros_with_same_feature_meta_default = l_flat_tangents_1_ = None
+
+        mul_tensor = torch.ops.aten.mul.Tensor(copy__default, getattr_l_self_fx_const_folded_attrs_1_);  copy__default = getattr_l_self_fx_const_folded_attrs_1_ = None
I want to see a test case for this, but I'm not sure that we actually bake things in . There's a graph break in the middle of linearize right now so what could be happening here is that every time one calls linearize it recompiles.

—
Reply to this email directly, [view it on GitHub](https://github.com/pytorch/pytorch/pull/123118#discussion_r1566449152), or [unsubscribe](https://github.com/notifications/unsubscribe-auth/A2NPQ3IGSQDHUHVYRGWTYADY5Q7YVAVCNFSM6AAAAABFSECIO6VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDAMBSGE4DINJVGM).
You are receiving this because you are subscribed to this thread.